### PR TITLE
Get current brightness, temperature and power state

### PR DESCRIPTION
**Pull request by hhaidar**
Hello,

Thanks again for writing this library @timrogers. I was looking for something to help me write a Stream Deck plugin and this seems like a good starting point.

I've added the following functions to the driver that people may find helpful:

* `getBrightnessInLumen`
* `getTemperatureInKelvin`
* `getPowerState`
* `toggle`

I've also updated the CLI to include a `litra-toggle` command and made the brightness/temp commands return the current value if no argument is passed into them. I did not change brightness percentage command or include a power state command.

Kudos to @abergmeier as [this thread ](https://github.com/kharyam/litra-driver/issues/22#issuecomment-1336168730) gave me what I needed. I managed to figure out the format of temperature values after a few hours of staring at the two bytes being returned (comments in the temp function).

Let me know if you need anything else from me to get this merged 🎉 

Ref #42

### CODE REVIEW

**Comment by timrogers** [comment]
@hhaidar Thanks! This is great ❤️

I'm actually already working on a PR locally for checking the current on/off state which I will push in the next couple of hours. 

I'll push that up and will take a look at this, and we can figure out how to move forward. Thank you for taking the time to contribute.

**Comment by timrogers** [comment]
@hhaidar Could you also tick the box to allow maintainers to update your PR? ☺️

File: src/driver.ts
Comment (timrogers) [inline src/driver.ts:139]: Can we rename this to `isOn`? I feel like that name fits nicely, and the API design of returning a boolean makes a lot of sense with that name.

```suggestion
export const isOn = (device: Device): boolean => {
```

After that change, we'd want to tweak the function documentation to align with it.

File: src/driver.ts
Comment (timrogers) [inline src/driver.ts:192]: Nice ❤️  And thanks for adding the comments here to set the context!

File: tests/driver.test.ts
Comment (timrogers) [inline tests/driver.test.ts:68]:
```suggestion
  it('sends the instruction to toggle the device on when it is off', () => {
```

File: src/cli/litra-brightness-lm.ts
Comment (timrogers) [inline src/cli/litra-brightness-lm.ts:1]: Can we update the `description` for the command too, e.g.:

> Manages the brightness of a Litra device. If a brightness argument is provided, it sets the device's brightness to the supplied value in Lumen. If no brightness argument is provided, it returns the current brightness in Lumen.

File: src/cli/litra-toggle.ts
Comment (timrogers) [inline src/cli/litra-toggle.ts:9]: For consistency with the docs of the other commands:

```suggestion
  .description('Toggles a Litra device on or off')
```

File: src/driver.ts
Comment (timrogers) [inline src/driver.ts:184]: What do these read functions return if the device is off?

File: README.md
Comment (timrogers) [inline README.md:114]:
```suggestion
  }, 5000));
```

File: README.md
Comment (timrogers) [inline README.md:125]:
```suggestion
// Turn your light on if it's currently off, and vice-versa
```

**Review by timrogers — CHANGES_REQUESTED** [review]
Really great work on this! I'm excited to get it merged ✨ I've left some small suggestions.

Would you also be able to update the commit messages to follow the conventions from `semantic-release`? I'd suggest the following messages:

* feat: add new `toggle`, `isOn`, `getBrightnessInLumen` and `getTemperatureInKelvin` functions
* chore(docs): update readme to reflect new functions
* feat: return current temperature and brightness respectively when `litra-temperature-k` and `litra-brightness-lm` are called without an argument
* chore: add missing power state tests

Once this is merged, I will follow up with improvement to the `litra-devices` CLI so that it returns the on/off state, temperature and brightness in its output. That will be really useful for https://github.com/timrogers/raycast-logitech-litra which depends on this package.

**Comment by timrogers** [comment]
Fixes #42.

File: src/cli/litra-brightness-lm.ts
Comment (hhaidar) [reply src/cli/litra-brightness-lm.ts:1]: Updated for both brightness and temperature.

File: src/cli/litra-brightness-lm.ts
Comment (hhaidar) [reply src/cli/litra-brightness-lm.ts:16]: 🙏

File: src/cli/litra-brightness-lm.ts
Comment (hhaidar) [reply src/cli/litra-brightness-lm.ts:26]: Yup, I tested all the commands locally before opening the PR.

File: src/driver.ts
Comment (hhaidar) [reply src/driver.ts:48]: Yeah if I `write` is sync then why not 🤷

File: src/driver.ts
Comment (hhaidar) [reply src/driver.ts:139]: The other functions have a verb as a prefix (like get, set, turn, etc..). Would it make more sense to call it `checkIfOn` or something similar for consistency? `isOn` doesn't sound like a function to me and if I were using the API I'd be a little surprised. Nonetheless it's your call and not a big deal, let me know!

File: src/driver.ts
Comment (hhaidar) [reply src/driver.ts:184]: They return the current value regardless if the light is powered on/off. Quite convenient.

File: src/driver.ts
Comment (timrogers) [reply src/driver.ts:139]: That's a good point about the normal use of verbs.\

If it's `getPowerState`, I'd expect it to return `'on'` or `'off'`, but really a boolean makes sense in this case.

I think I'd still prefer to go for `isOn`, if you don't mind. It just feels like the most natural option to me!

**Comment by hhaidar** [comment]
@timrogers thank you for the quick feedback, I will rebase and squash into 1 comment at the end if that's alright

File: src/cli/litra-temperature-k.ts
Comment (hhaidar) [reply src/cli/litra-temperature-k.ts:1]: Done

**Comment by timrogers** [comment]
> @timrogers thank you for the quick feedback, I will rebase and squash into 1 comment at the end if that's alright

Sure! Just ping me once you're ready for re-review.

**Comment by hhaidar** [comment]
@timrogers I'm actually ready now, all issues should be resolved I think.

**Review by timrogers — APPROVED** [review]

**Comment by jaaortega** [comment]
Hello friends, I first want to say I am grateful for all the hard work in reverse engineering the litra api! I am a windows user who has been taking tips from all this conversation to construct python scripts that will increment/decrement temperature and brightness on the stream deck.

I was excited to see that someone had figured out a way to get the current state of the device as my current scripts utilize reading/writing to a text file as a psuedo-database for storing the current temp or brightness. I wish I could instead read the current status of the device and increment/decrement brightness/temp 

I wrote the following python script which will do such a thing for brightness and temp, In the example for brightness, what it does is read in a txt file which has the current brightness as a hex value, then it increments/decrements the hex value by a "1C" hex step and writes the new value back to a file. it then takes that new hex value and processes it to be injected into a hidapitester.exe command which sends the correct hex output string to make the light adjust accordingly.

The script has a static path for the brightness.txt file and uses a command that specifically targets one of two of my litra lights:

import os
import subprocess


def adjust_and_invoke_hidapitester(increment):
    # Set the file path
    file_path = "C:/Users/jortega/Documents/Stream Deck Bashes/Litra Stuff/left_current_brightness.txt"

    # Read the current brightness value from file (as hex)
    with open(file_path, 'r') as file:
        current_brightness = file.read().strip()

    # Increment or decrement the hex value by "1C"
    if increment:
        current_brightness = hex(int(current_brightness, 16) + 0x1C)[2:].zfill(4)
    else:
        current_brightness = hex(int(current_brightness, 16) - 0x1C)[2:].zfill(4)

    # Validate the current brightness range
    if int(current_brightness, 16) < 0x1C:
        current_brightness = "001C"
    elif int(current_brightness, 16) > 0xFF:
        current_brightness = "00FF"

    # Write the new current brightness back to file
    with open(file_path, 'w') as file:
        file.write("0x" + current_brightness)

    # Split the base 16 value into two 2-character strings
    split1 = current_brightness[:2]
    split2 = current_brightness[2:]

    # Construct the command for hidapitester.exe
    command = (
        'hidapitester.exe --vidpid 046d/c900 --usagePage ff43 --open-path '
        '"\\\\?\\HID#VID_046D&PID_C900&Col02#8&1e2a4b18&0&0001#{4d1e55b2-f16f-11cf-88cb-001111000030}" '
        f'--send-output 0x11,0xFF,0x04,0x4F,0x{split1},0x{split2}'
    )

    # Execute the command with subprocess
    subprocess.run(command, shell=True)


def main():
    # Set the increment flag to True for incrementing the brightness
    increment = True

    # Call the function to adjust the brightness and invoke hidapitester.exe
    adjust_and_invoke_hidapitester(increment)


if __name__ == '__main__':
    main()


So I directly run this script on an System > Open command on Streamdeck button and it works. If you want to decrement, you just change the value of increment in the main() function to False. Temperature follows the same idea of having a base 16 value except instead of being a base 10 value between 20-250, temperature is 2700-6500 (source: https://www.reddit.com/r/LogitechG/comments/sacz2x/comment/ixtez4q/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)

**Pull request merged**
